nautobot_netbox_importer/diffsync/models/locations.py

```diff
@@ -3,119 +3,111 @@
 
         if parent:
-            result = field.wrapper.get_pk_from_uid(parent)
-            field.wrapper.add_reference(result, field.wrapper)
+            result = wrapper.get_pk_from_uid(parent)
+            wrapper.add_reference(wrapper, result)
         elif site:
             result = site_wrapper.get_pk_from_uid(site)
-            site_wrapper.add_reference(result, field.wrapper)
+            wrapper.add_reference(site_wrapper, result)
         else:
             return
 
         setattr(target, field.nautobot.name, result)
 
-    field.set_importer(importer)
+    field.set_nautobot_field(field.name)
+    field.set_importer(location_parent_importer)
+    field.handle_sibling("site", field.nautobot.name)
 
 
 def _define_site_group(field: SourceField) -> None:
-    site_group_wrapper = field.wrapper.adapter.wrappers["dcim.sitegroup"]
-    location_type_wrapper = field.wrapper.adapter.wrappers["dcim.locationtype"]
+    wrapper = field.wrapper
+    site_group_wrapper = wrapper.adapter.wrappers["dcim.sitegroup"]
+    location_type_wrapper = wrapper.adapter.wrappers["dcim.locationtype"]
     site_type_uid = location_type_wrapper.get_pk_from_uid("Site")
 
     field.set_nautobot_field("location_type")
 
-    def importer(source: RecordData, target: DiffSyncBaseModel) -> None:
+    def site_group_importer(source: RecordData, target: DiffSyncBaseModel) -> None:
         group = source.get(field.name, None)
         if group:
             result = site_group_wrapper.get_pk_from_uid(group)
-            site_group_wrapper.add_reference(result, field.wrapper)
+            wrapper.add_reference(site_group_wrapper, result)
         else:
             result = site_type_uid
-            location_type_wrapper.add_reference(result, field.wrapper)
+            wrapper.add_reference(location_type_wrapper, result)
 
         setattr(target, field.nautobot.name, result)
 
-    field.set_importer(importer)
+    field.set_importer(site_group_importer)
 
 
 def setup(adapter: SourceAdapter) -> None:
     """Setup locations for NetBox importer."""
+
+    def forward_references(wrapper: SourceModelWrapper, references: SourceReferences) -> None:
+        for uid, wrappers in references.items():
+            instance = wrapper.get_or_create(uid, fail_missing=True)
+            location_type_uid = getattr(instance, "location_type_id")
+            for item in wrappers:
+                item.add_reference(location_type_wrapper, location_type_uid)
+
     options = getattr(adapter, "options", {})
     sitegroup_parent_always_region = getattr(options, "sitegroup_parent_always_region", False)
 
-    location_type_wrapper = adapter.configure_model("dcim.locationtype")
+    location_type_wrapper = adapter.configure_model("dcim.LocationType")
 
     region_type_uid = location_type_wrapper.cache_record(
         {
             "id": "Region",
             "name": "Region",
             "nestable": True,
-            "content_types": None,
```

Comment:
why are no needed any more?

Reply:
The same as `status` fields, `content_types` are auto added based on existence in Nautobot.

https://github.com/nautobot/nautobot-app-netbox-importer/pull/123/files#diff-6b479358489e04d98598db52da0a5cce8c2636d21202d851e16589d38820e84eR103